__tests__/commands/upgrade.js

```diff
@@ -102,6 +102,70 @@ test.concurrent('upgrades from fixed version to latest', (): Promise<void> => {
   });
 });
 
+test.concurrent('upgrades to latest matching package.json semver when no package name passed', (): Promise<void> => {
+  return runUpgrade([], {}, 'range-to-latest', async (config): ?Promise<void> => {
+    const lockfile = explodeLockfile(await fs.readFile(path.join(config.cwd, 'yarn.lock')));
+    const pkg = await fs.readJson(path.join(config.cwd, 'package.json'));
+    const lockEntryIndex = lockfile.indexOf('left-pad@<=1.1.1:');
+
+    expect(lockEntryIndex).toEqual(0);
+    expect(lockfile[lockEntryIndex + 1]).toContain('1.1.1');
+    expect(pkg.dependencies).toEqual({'left-pad': '<=1.1.1'});
+  });
+});
+
+test.concurrent('--latest upgrades to latest ignoring package.json when no package name passed', (): Promise<void> => {
+  return runUpgrade([], {latest: true}, 'range-to-latest', async (config): ?Promise<void> => {
+    const lockfile = explodeLockfile(await fs.readFile(path.join(config.cwd, 'yarn.lock')));
+    const pkg = await fs.readJson(path.join(config.cwd, 'package.json'));
+    const lockEntryIndex = lockfile.indexOf('left-pad@^1.1.3:');
+
+    expect(lockEntryIndex).toEqual(0);
+    expect(lockfile.indexOf('left-pad@<=1.1.1:')).toEqual(-1);
+    expect(lockfile[lockEntryIndex + 1]).toContain('1.1.3');
+    expect(pkg.dependencies).toEqual({'left-pad': '^1.1.3'});
```

Comment:
You shouldn't use `left-pad` for this test, otherwise it will break if they release a new version. I think you can use `yarn-test` (cc @kittens ?)